lib/cocoapods/installer/analyzer.rb

```diff
@@ -163,12 +163,12 @@ def generate_podfile_state
         end
       end
 
-      # Updates the source repositories unless the config indicates to skip it.
+      # Updates the git source repositories unless the config indicates to skip it.
       #
       def update_repositories_if_needed
         unless config.skip_repo_update?
           UI.section 'Updating spec repositories' do
-            sources.each { |source| SourcesManager.update(source.name) }
+            sources.each { |source| SourcesManager.update(source.name) if SourcesManager.git_repo?(source.repo) }
```

Comment:
Perhaps it makes sense to introduce a new method on the source manager, `can_update?` or something similar. Since updating it not actually specific to git updating. Other types of sources provided by a plugin may also support updating. Such as your very own SVN plugin: https://github.com/clarkda/cocoapods-repo-svn

The default implementation returns false, whereas the git one will override and return true.

That, or even update just raises `NoMethodError`/`NotImplementedError`. What do you think @segiddins @clarkda?